### DISCUSSION TITLE
Fine tuning Yolov8s For just a person

### DISCUSSION THREAD

**Discussion by memoriesbytalha** [ultralytics/ultralytics#14331] [Q&A]
Hey there Yolo Fam!
I am sorry i've been asking alot of questions.
I want to finetune my yolo model to just detect the person , I can hard code by passing the parameter as classes_to_count = [0] but I want to freeze all the the rest classes and fine tune just this one.

**Comment by pderrenger** [comment]
@memoriesbytalha hello!

No worries at all, we're here to help! To fine-tune YOLOv8 to detect only the "person" class, you can indeed freeze the other classes during training. Here’s how you can achieve this:

1. **Modify the Dataset**: Ensure your dataset only includes annotations for the "person" class. This will help the model focus solely on detecting persons.

2. **Freeze Layers**: You can freeze certain layers of the model to retain their pre-trained weights while fine-tuning the rest. This can be done by specifying the `freeze` parameter during training.

Here’s a code snippet to illustrate this:

```python
from ultralytics import YOLO

# Load the pre-trained model
model = YOLO('yolov8s.pt')

# Train the model, freezing the first N layers
model.train(data='path/to/your/dataset.yaml', epochs=50, freeze=10, classes=[0])
```

In this example:
- `freeze=10` freezes the first 10 layers of the model.
- `classes=[0]` ensures that only the "person" class is considered during training.

3. **Custom Training Script**: If you need more control, you can create a custom training script to handle specific requirements.

For more detailed guidance, you can refer to our [documentation on training](https://docs.ultralytics.com/modes/train).

Feel free to reach out if you have more questions. Happy training! 🚀
